src/tacacs/pam/0010-handle-bad-password-set-by-sshd.patch

```diff
@@ -0,0 +1,99 @@
+From eb7f56e4a75050091916841d2d9daec646a4210a Mon Sep 17 00:00:00 2001
+From: Renuka Manavalan <remanava@microsoft.com>
+Date: Fri, 29 Oct 2021 21:40:55 +0000
+Subject: [PATCH] handle bad password set by sshd
+
+---
+ pam_tacplus.c |  7 +++++++
+ support.c     | 39 ++++++++++++++++++++++++++++++++++++++-
+ support.h     |  1 +
+ 3 files changed, 46 insertions(+), 1 deletion(-)
+
+diff --git a/pam_tacplus.c b/pam_tacplus.c
+index d57657a..2f1ed46 100644
+--- a/pam_tacplus.c
++++ b/pam_tacplus.c
+@@ -248,6 +248,13 @@ int pam_sm_authenticate (pam_handle_t * pamh, int flags,
+         return PAM_CRED_INSUFFICIENT;
+     }
+ 
++    if (validate_not_sshd_bad_pass(pass) != PAM_SUCCESS) {
++        syslog(LOG_ERR, "auth fail: Password incorrect");
++        bzero(pass, strlen (pass));
```

Comment:
The bzero function been deprecated because the compiler may remove the code: https://man7.org/linux/man-pages/man3/bzero.3.html

However this is not a bug, because when code run to here, the 'pass' only contains SSHD_BAD_PASS, so no risk to leak user secret.

Reply:
The code is currently using bzero in other places too. That is how, I put it.
Does it mean, we patch other occurrences too ?

Reply:
Searched across and fixed all.